aten/src/ATen/native/mobile/cpu/internal/Add.cpp

```diff
@@ -0,0 +1,244 @@
+#include <ATen/native/mobile/cpu/Engine.h>
+
+#ifdef USE_XNNPACK
+
+#include <ATen/native/mobile/cpu/internal/Allocator.h>
+#include <ATen/native/mobile/cpu/internal/Common.h>
+
+namespace at {
+namespace native {
+namespace mobile {
+namespace cpu {
+namespace internal {
+namespace add {
+
+struct Context final {
+  Operator add_op;
+
+  static constexpr float kMin = -std::numeric_limits<float>::infinity();
+  static constexpr float kMax = std::numeric_limits<float>::infinity();
+};
+
+namespace {
+
+// Supports NHWC and NCHW FP32 adds.
+
+bool available(
+    const int64_t input1_channels,
+    const int64_t input2_channels,
+    const int64_t output_channels,
+    const Scalar output_min,
+    const Scalar output_max) {
+         // Mobile
+  return mobile::cpu::available() &&
+         // Input1
+         (input1_channels > 0) &&
+         // Input2
+         (input2_channels > 0) &&
+         // Input1 == Input2
+         (input1_channels == input2_channels) &&
+         // Output
+         (output_channels > 0) &&
+         // Input == Output
+         (input1_channels == output_channels) &&
+         // Output Min / Max
+         (output_min.isIntegral(true) || output_min.isFloatingPoint()) &&
+         (output_max.isIntegral(true) || output_max.isFloatingPoint()) &&
+         (output_max.to<float>() > output_min.to<float>());
+}
+
+Context create(
+    const int64_t input1_channels,
+    const int64_t input2_channels,
+    const int64_t output_channels,
+    const Scalar output_min,
+    const Scalar output_max) {
+  TORCH_INTERNAL_ASSERT(
+      available(input1_channels, input2_channels, output_channels, output_min, output_max),
+      "mobile::cpu::add not available!");
+
+  xnn_operator_t add_op{};
```

Comment:
Can you tell me a little more about what is stored in `xnn_operator_t`?

Reply:
https://github.com/AshkanAliabadi/XNNPACK/blob/1465014503787e1a2f0806a729d511a5c49bb231/src/xnnpack/operator.h#L162

Reply:
Yeah you found it.  xnn_operator_t is an opaque type - a pointer to xnn_operator.